django_mongodb/functions.py

```diff
@@ -57,6 +72,23 @@
 }
 
 
+def aggregate(self, compiler, connection, **extra_context):  # noqa: ARG001
+    if self.filter:
+        copy = self.copy()
+        copy.filter = None
+        source_expressions = copy.get_source_expressions()
+        condition = When(self.filter, then=source_expressions[0])
+        copy.set_source_expressions([Case(condition)] + source_expressions[1:])
+        node = copy
+    else:
+        node = self
+    lhs_mql = process_lhs(node, compiler, connection)
+    operator = MONGO_AGGREGATIONS.get(self.__class__)
```

Comment:
This could be `MONGO_AGGREGATIONS.get(self.__class__, self.function.lower())` (similar to the default fallback for database functions) so that Avg, Max, Min, and Sum could be removed from `MONGO_AGGREGATIONS`.